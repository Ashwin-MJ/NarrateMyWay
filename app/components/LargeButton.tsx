import { Component } from 'react';

import * as React from 'react';
import { Pressable, Text } from 'react-native';
import styles from './styles/LargeButton.style';

class LargeButton extends Component {
  render() {
    return (
      <Pressable
        style={styles.largeButtonContainer}
        android_ripple={{ color: '#fff' }}
        onPress={() => {
          // Triggered when the user taps on the button
          console.log('Test onPress');
        }}
        onPressOut={() => {
          // Triggered when user removes their finger
          console.log('Test onPressOut');
        }}
        onLongPress={() => {
          // Triggered when the user presses and holds for longer than 500ms
          // Can customise the duration by setting delayLongPress
          console.log('Test onLongPress');
        }}
        accessible={true}
        accessibilityLabel={this.props.accessibilityLabel}
      >
        <Text style={styles.largeButtonText} adjustsFontSizeToFit>
          {this.props.children}
        </Text>
      </Pressable>
    );
  }
}

export default LargeButton;
