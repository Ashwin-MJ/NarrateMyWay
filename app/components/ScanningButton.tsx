import { Component } from 'react';

import * as React from 'react';
import { Pressable, Text } from 'react-native';
import styles from './styles/ScanningButton.style';

import * as Speech from 'expo-speech';
import DefaultColors from '../constants/DefaultColors';

class ScanningButton extends Component {
  render() {
    return (
      <Pressable
        style={styles.buttonContainer}
        android_ripple={DefaultColors.rippleColor}
        onPress={() => {
          Speech.speak('Scanning for beacons');
        }}
        accessible={true}
        accessibilityLabel={this.props.accessibilityLabel}
      >
        <Text style={styles.buttonText} adjustsFontSizeToFit>
          Scanning...
        </Text>
      </Pressable>
    );
  }
}

export default ScanningButton;
