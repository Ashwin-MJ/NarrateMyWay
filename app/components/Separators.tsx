import { Component } from 'react';

import * as React from 'react';
import { View } from 'react-native';
import styles from './styles/Separators.style';

class HorizontalSeparator extends Component {
  render() {
    return <View style={styles.horizontalSeparator} />;
  }
}

class VerticalSeparator extends Component {
  render() {
    return <View style={styles.verticalSeparator} />;
  }
}

export { HorizontalSeparator, VerticalSeparator };
