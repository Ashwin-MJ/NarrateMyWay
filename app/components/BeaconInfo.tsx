import { Component } from 'react';

import * as React from 'react';
import { Pressable, Text } from 'react-native';
import styles from './styles/BeaconInfo.style';
import { Ionicons } from '@expo/vector-icons';
import { View } from 'react-native';
import { HorizontalSeparator, VerticalSeparator } from './Separators';

import * as Speech from 'expo-speech';

class BeaconInfo extends Component {
  render() {
    return (
      <View style={styles.beaconInfoContainer}>
        <TypeText title={this.props.type}></TypeText>
        <HorizontalSeparator />
        <View style={styles.placeContainer}>
          <PlaceIcon></PlaceIcon>
          <VerticalSeparator />
          <PlaceText title={this.props.place}></PlaceText>
        </View>
      </View>
    );
  }
}

const TypeText = ({ title }) => (
  <Pressable
    style={styles.typeContainer}
    android_ripple={{ color: '#fff' }}
    accessible={true}
    onPress={() => {
      Speech.speak('A point of interest has been located near you');
    }}
  >
    <Text style={styles.typeText} numberOfLines={1} adjustsFontSizeToFit>
      {title}
    </Text>
  </Pressable>
);

const PlaceIcon = () => (
  <Ionicons
    name="cafe"
    style={styles.placeIcon}
    size={80}
    accessible={true}
    accessibilityLabel="Icon of a cafe"
    accessibilityHint="test"
    adjustsFontSizeToFit
  />
);

const PlaceText = ({ title }) => (
  <Pressable
    style={styles.placeTextContainer}
    android_ripple={{ color: '#fff' }}
    accessible={true}
    onPress={() => {
      Speech.speak('A cafe has been located near you');
    }}
  >
    <Text style={styles.placeText} numberOfLines={1} adjustsFontSizeToFit>
      {title}
    </Text>
  </Pressable>
);

export default BeaconInfo;
