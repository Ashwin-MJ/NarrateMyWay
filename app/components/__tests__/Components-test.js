import * as React from 'react';
import renderer from 'react-test-renderer';
import LargeButton from '../LargeButton';
import BeaconInfo from '../BeaconInfo';
import ScanningButton from '../ScanningButton';
import { HorizontalSeparator, VerticalSeparator } from '../Separators';

// LargeButton tests
it(`LargeButton renders correctly`, () => {
  const tree = renderer
    .create(
      <LargeButton accessibilityLabel="Test accessibility label">
        Tap to repeat
      </LargeButton>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`LargeButton text and accessibility correct`, () => {
  const largeButton = (
    <LargeButton accessibilityLabel="Test accessibility label">
      Tap to repeat
    </LargeButton>
  );

  expect(largeButton.props.accessibilityLabel).toBe('Test accessibility label');
  expect(largeButton.props.children).toBe('Tap to repeat');
});

// Separators tests
it(`HorizontalSeparator renders correctly`, () => {
  const tree = renderer.create(<HorizontalSeparator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`VerticalSeparator renders correctly`, () => {
  const tree = renderer.create(<VerticalSeparator />).toJSON();
  expect(tree).toMatchSnapshot();
});

// BeaconInfo tests
it(`BeaconInfo renders correctly`, () => {
  const tree = renderer
    .create(<BeaconInfo type="Test Type" place="Test Place" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`BeaconInfo type and place correct`, () => {
  const beaconInfo = <BeaconInfo type="Test Type" place="Test Place" />;

  expect(beaconInfo.props.type).toBe('Test Type');
  expect(beaconInfo.props.place).toBe('Test Place');
});

// ScanningButton tests
it(`ScanningButton renders correctly`, () => {
  const tree = renderer
    .create(<ScanningButton accessibilityLabel="Scanning" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it(`ScanningButton type and place correct`, () => {
  const scanningButton = <ScanningButton accessibilityLabel="Scanning" />;

  expect(scanningButton.props.accessibilityLabel).toBe('Scanning');
});
