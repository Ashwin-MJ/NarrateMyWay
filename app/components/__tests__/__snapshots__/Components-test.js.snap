// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeaconInfo renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#ff9900",
        "borderRadius": 50,
        "elevation": 8,
        "paddingHorizontal": 30,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 36,
          "fontWeight": "bold",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Test Type
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text />
    <View
      style={
        Object {
          "marginVertical": 10,
          "width": 20,
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#009688",
          "borderRadius": 50,
          "elevation": 8,
          "paddingHorizontal": 70,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 42,
            "fontWeight": "bold",
            "textTransform": "uppercase",
          }
        }
      >
        Test Place
      </Text>
    </View>
  </View>
</View>
`;

exports[`HorizontalSeparator renders correctly 1`] = `
<View
  style={
    Object {
      "height": 1,
      "marginVertical": 10,
      "width": "80%",
    }
  }
/>
`;

exports[`LargeButton renders correctly 1`] = `
<View
  accessibilityLabel="Test accessibility label"
  accessible={true}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#093f74",
      "elevation": 8,
      "height": "30%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 42,
        "fontWeight": "bold",
        "justifyContent": "center",
        "textAlign": "center",
        "textTransform": "uppercase",
      }
    }
  >
    Tap to repeat
  </Text>
</View>
`;

exports[`MainScreen renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessibilityLabel="Tap here to repeat the previous audio output"
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#093f74",
        "elevation": 8,
        "height": "30%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 42,
          "fontWeight": "bold",
          "justifyContent": "center",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Tap to repeat
    </Text>
  </View>
  <View
    style={
      Object {
        "height": 1,
        "marginVertical": 10,
        "width": "80%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ff9900",
          "borderRadius": 50,
          "elevation": 8,
          "paddingHorizontal": 30,
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontSize": 36,
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Point of Interest 
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text />
      <View
        style={
          Object {
            "marginVertical": 10,
            "width": 20,
          }
        }
      />
      <View
        accessible={true}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#009688",
            "borderRadius": 50,
            "elevation": 8,
            "paddingHorizontal": 70,
            "paddingVertical": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 42,
              "fontWeight": "bold",
              "textTransform": "uppercase",
            }
          }
        >
          Cafe 
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "height": 1,
        "marginVertical": 10,
        "width": "80%",
      }
    }
  />
  <View
    accessibilityLabel="Tap here for more information"
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#093f74",
        "elevation": 8,
        "height": "30%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "#fff",
          "fontSize": 42,
          "fontWeight": "bold",
          "justifyContent": "center",
          "textAlign": "center",
          "textTransform": "uppercase",
        }
      }
    >
      Tap for more info
    </Text>
  </View>
</View>
`;

exports[`VerticalSeparator renders correctly 1`] = `
<View
  style={
    Object {
      "marginVertical": 10,
      "width": 20,
    }
  }
/>
`;
